src/simple_engine.rs

```diff
@@ -0,0 +1,106 @@
+use crate::description::LevelDescription;
+use crate::game_state::CellState;
+use crate::grid::Grid;
+use crate::play_engine::PlayEngine;
+
+pub struct SimpleEngine {
+
+}
+impl PlayEngine for SimpleEngine {
+    fn play_one_turn(&self, level_description: &LevelDescription, grid: &mut dyn Grid) -> bool {
+        for row_description in level_description.rows.iter().enumerate() {
+            if !row_description.1.iter().all(|x| x.1) {
+                let mut begin_offset = 0;
+                let mut end_offset = 0;
+
+                for i in 0..grid.width() {
+                    if grid.get(i, row_description.0) == CellState::Crossed {
+                        begin_offset += 1;
+                    }
+                    else {
+                        break;
+                    }
+                }
+
+                for i in (0..grid.width()).rev() {
```

Comment:
Can be rewritten using take while, rewrite or leave a TODO comment
```suggestion
                for i in (0..grid.width()).rev() { // TODO: rewrite using take-while
```